internal/gatewayapi/translator.go

```diff
@@ -151,12 +152,13 @@ type portListeners struct {
 	hostnames map[string]int
 }
 
-func (t *Translator) ProcessListeners(gateways []*GatewayContext, xdsIR *ir.Xds, infraIR *ir.Infra, resources *Resources) {
+func (t *Translator) ProcessListeners(gateways []*GatewayContext, xdsIR XdsIRMap, infraIR InfraIRMap, resources *Resources) {
 	portListenerInfo := map[v1beta1.PortNumber]*portListeners{}
 
 	// Iterate through all listeners and collect info about protocols
 	// and hostnames per port.
 	for _, gateway := range gateways {
+
```

Comment:
I think this section will need some adjustments as well since it's looking for port/protocol/hostname conflicts between Listeners across all Gateways, which shouldn't be necessary with this approach, only need to look for conflicts between Listeners on each Gateway individually.

Reply:
thanks for catching this, fixed with https://github.com/envoyproxy/gateway/pull/404/commits/3435da1e8aa3f01c7e1eadd0d4841adec769e410